subworkflows/local/germline_variant_calling.nf

```diff
@@ -6,369 +6,383 @@ include { BGZIP as BGZIP_DEEPVARIANT_GVCF             } from '../../modules/loca
     // Deepvariant: https://github.com/google/deepvariant/issues/510
 
-    if (tools.contains('deepvariant')) {
-        DEEPVARIANT(
-            cram_recalibrated_intervals,
-            fasta,
-            fasta_fai)
-        ch_versions = ch_versions.mix(DEEPVARIANT.out.versions)
-
-        if(no_intervals){
-            TABIX_DEEPVARIANT_VCF(DEEPVARIANT.out.vcf)
-            TABIX_DEEPVARIANT_GVCF(DEEPVARIANT.out.gvcf)
+    DEEPVARIANT(
+        cram_recalibrated_intervals,
+        fasta,
+        fasta_fai)
 
-            deepvariant_vcf_gz = DEEPVARIANT.out.vcf
-            deepvariant_gvcf_gz = DEEPVARIANT.out.gvcf
+    // Only when no intervals
+    TABIX_DEEPVARIANT_VCF(DEEPVARIANT.out.vcf)
+    TABIX_DEEPVARIANT_GVCF(DEEPVARIANT.out.gvcf)
 
-            ch_versions = ch_versions.mix(TABIX_DEEPVARIANT_VCF.out.versions)
-            ch_versions = ch_versions.mix(TABIX_DEEPVARIANT_GVCF.out.versions)
-        }else{
-            BGZIP_DEEPVARIANT_VCF(DEEPVARIANT.out.vcf)
-            BGZIP_DEEPVARIANT_GVCF(DEEPVARIANT.out.gvcf)
+    // Only when using intervals
+    BGZIP_DEEPVARIANT_VCF(DEEPVARIANT.out.vcf)
+    BGZIP_DEEPVARIANT_GVCF(DEEPVARIANT.out.gvcf)
 
-            BGZIP_DEEPVARIANT_VCF.out.vcf.map{ meta, vcf ->
+    CONCAT_VCF_DEEPVARIANT(
+        BGZIP_DEEPVARIANT_VCF.out.vcf
+            .map{ meta, vcf ->
                 new_meta = meta.clone()
                 new_meta.id = new_meta.sample
                 [new_meta, vcf]
-            }.groupTuple(size: num_intervals)
-            .set{deepvariant_vcf_to_concat}
+            }.groupTuple(size: num_intervals),
+        fasta_fai,
+        intervals_bed_combine_gz)
 
-            BGZIP_DEEPVARIANT_GVCF.out.vcf.map{ meta, vcf ->
+    CONCAT_GVCF_DEEPVARIANT(
+        BGZIP_DEEPVARIANT_GVCF.out.vcf
+            .map{ meta, vcf ->
                 new_meta = meta.clone()
                 new_meta.id = new_meta.sample
                 [new_meta, vcf]
-            }.groupTuple(size: num_intervals)
-            .set{deepvariant_gvcf_to_concat}
-
-            CONCAT_VCF_DEEPVARIANT(deepvariant_vcf_to_concat, fasta_fai, intervals_bed_combine_gz)
-            CONCAT_GVCF_DEEPVARIANT(deepvariant_gvcf_to_concat, fasta_fai, intervals_bed_combine_gz)
-
-            deepvariant_vcf_gz = CONCAT_VCF_DEEPVARIANT.out.vcf
-            deepvariant_gvcf_gz = CONCAT_GVCF_DEEPVARIANT.out.vcf
-
-            ch_versions = ch_versions.mix(BGZIP_DEEPVARIANT_VCF.out.versions)
-            ch_versions = ch_versions.mix(CONCAT_VCF_DEEPVARIANT.out.versions)
-        }
+            }.groupTuple(size: num_intervals),
+        fasta_fai,
+        intervals_bed_combine_gz)
 
-        deepvariant_vcf = deepvariant_vcf.mix(deepvariant_vcf_gz, deepvariant_gvcf_gz)
-    }
+    deepvariant_vcf = channel.empty().mix(
+        CONCAT_GVCF_DEEPVARIANT.out.vcf,
+        CONCAT_VCF_DEEPVARIANT.out.vcf,
+        DEEPVARIANT.out.gvcf,
```

Comment:
wouldn't this add the intervalled vfcs even when they are merged? Because it is run in any case

Reply:
Oh yeah, I meant to use the TABIX output there...